src/components/AutoComplete/AutoComplete.stories.js

```diff
@@ -0,0 +1,96 @@
+
+AutoCompleteStories.addDecorator(withKnobs);
+
+AutoCompleteStories.addWithInfo('Basic Example', () => {
+  const highlightOnlyResult = boolean('highlight Only Result', true);
+  const clearButton = boolean('Clear Button', true);
+  const multiple = boolean('Multiple Selection', false);
+  const allowNew = boolean('Allow Adding New Items', false);
+  const selectHintOnEnter = boolean('select Hint On Enter', true);
+  const minLength = number("Min Length to Autocomplete", 0);
+  const bsSize = select('Size', [undefined, 'default', 'small', 'large']);
+  const align = select('Dropdown Alignment', [undefined, 'justify', 'left', 'right']);
+  const props = { bsSize, align, clearButton, multiple, allowNew, options, minLength, selectHintOnEnter, highlightOnlyResult }
+
+  return (
+    <div className="container">
+      <Col>
+        <br></br>
+        <br></br>
+        <br></br>
+        <h2>Basic Example</h2>
+        <AutoComplete
+          {...props}
+          labelKey="name"
+          placeholder="Choose a state..."
+        />
+      </Col>
+    </div>
+  );
+});
+
+AutoCompleteStories.addWithInfo('Use With Forms', () => {
+  const submitFormOnEnter = boolean('Submit Form On Enter', true);
+  return (
+    <div className="container">
+      <br></br>
+      <br></br>
+      <br></br>
+      <h2>Submit Form On 'Enter'</h2>
+      <Form horizontal onSubmit={(e) => alert('Form submitted!')}>
+        <Col sm={10}>
+          <FormGroup>
+            <AutoComplete
+              submitFormOnEnter={submitFormOnEnter}
+              labelKey="name"
+              options={options}
+              placeholder="Choose a state..."
+            />
+          </FormGroup>
+        </Col>
+        <Col sm={2}>
+          <FormGroup>
+            <button className="btn-secondary" type="submit">
+              Submit
+            </button>
+          </FormGroup>
+        </Col>
+      </Form>
+    </div>
+  );
+});
+
+AutoCompleteStories.addWithInfo('Using Async Calls', () => {
+
+  return (
+    <div className="container">
+      <br></br>
+      <br></br>
+      <br></br>
+      <h2>Search a Forker</h2>
+      <AsyncAutoComplete
+        labelKey="login"
+        minLength={0}
+        onSearch={() => mockData.map(item => item.owner)}
```

Comment:
I think it would be nice to actually make an async call here.  Either by mocking the data to be a function that returns a promise, by actually calling Github's API, or by calling a placeholder service.  That would be a better test/example that matches consumption.

Reply:
I actually used Github's API until they locked me out for too many calls.. Think that using a promise might be also a good example.